src/loaders/ncbi_downloader/ncbi_downloader_helper.py

```diff
@@ -0,0 +1,196 @@
+                host.chdir(gene_dir_name)
+                gene_file_list = host.listdir(host.curdir)
+
+                for gene_file_name in gene_file_list:
+                    # file has target extensions but doesn't contain exclude name substring
+                    if any([gene_file_name.endswith(ext) for ext in target_file_ext]) and all(
+                            [substring not in gene_file_name for substring in exclude_name_substring]):
+                        result_file_path = os.path.join(download_dir, gene_file_name)
+                        if overwrite or not os.path.exists(result_file_path):
+                            host.download(gene_file_name, result_file_path)
+
+            success = True
+        except Exception as e:
+            print(f'Error:\n{e}\nfrom attempt {attempts + 1}.\nTrying to rerun.')
+            attempts += 1
+
+    if not success:
+        raise ValueError(f'Download Failed for {gene_id} after {max_attempts} attempts!')
+
+
+def download_genome_files(
+        gene_ids: list[str],
+        target_file_ext: list[str],
+        exclude_name_substring: list[str],
+        root_dir: str,
+        overwrite: bool = False
+) -> list[str]:
+    """
+    Download genome files from NCBI FTP server
+
+    gene_ids: genome ids that parsed from GTDB metadata file (e.g. GB_GCA_000016605.1, RS_GCF_000968435.2).
+    target_file_ext: download only files that match given extensions.
+    exclude_name_substring: exclude files that contain given substrings in their names.
+    root_dir: root directory for the collections project.
+    overwrite: overwrite existing files if True.
+    """
+
+    print(f'start downloading {len(gene_ids)} genome files')
+
+    failed_ids = list()
+
+    result_dir = get_work_dir(root_dir)
+
+    counter = 1
+    for gene_id in gene_ids:
+        if counter % 5000 == 0:
+            print(f"{round(counter / len(gene_ids), 4) * 100}% finished at {datetime.now()}")
+
+        download_dir = os.path.join(result_dir, gene_id)
+        os.makedirs(download_dir, exist_ok=True)
+
+        try:
+            _download_genome_file(download_dir, gene_id, target_file_ext, exclude_name_substring, overwrite=overwrite)
+        except Exception as e:
+            print(e)
+            failed_ids.append(gene_id)
+
+        counter += 1
+
+    if failed_ids:
+        print(f'Failed to download {failed_ids}')
+
+    return failed_ids
+
+
+def download_genome_files_in_parallel(
+        root_dir: str,
+        genome_ids: list[str],
+        target_file_ext: list[str],
+        exclude_name_substring: list[str],
+        system_utilization: float,
+        threads: int = None,
+        overwrite: bool = False
+) -> list[list[str]]:
+    """
+    Download genome files from NCBI FTP server in parallel using multiprocessing
+
+    root_dir: root directory for the collections project.
+    genome_ids: genome ids that parsed from GTDB metadata file (e.g. GB_GCA_000016605.1, RS_GCF_000968435.2).
```

Comment:
```suggestion
    genome_ids: NCBI genome ids to download (e.g. GB_GCA_000016605.1, RS_GCF_000968435.2).
```

May or may not be parsed from the GTDB file in future

Reply:
Same comment for the docs in other 2 functions

Reply:
Updated!

Reply:
:+1: